cmd/fossa/init.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+
+	logging "github.com/op/go-logging"
+	"github.com/urfave/cli"
+
+	"github.com/fossas/fossa-cli/builders"
+	config "github.com/fossas/fossa-cli/config"
+)
+
+var initLogger = logging.MustGetLogger("init")
+
+func initCmd(c *cli.Context) {
+	conf, err := config.New(c)
+	if err != nil {
+		initLogger.Fatalf("Could not load configuration: %s", err.Error())
+	}
+
+	if err := doInit(&conf, c.Bool("force"), c.Bool("include-all")); err != nil {
+		initLogger.Fatalf("Error initializing: %s", err.Error())
+	}
+
+	if !c.Bool("skip-write") {
+		if err := config.WriteConfigFile(&conf); err != nil {
+			initLogger.Fatalf("Error writing config: %s", err.Error())
+		}
+
+		initLogger.Warningf("Config for %d modules written to `%s`.", len(conf.Modules), conf.ConfigFilePath)
+	}
+
+	fmt.Println("`fossa` is initialized")
+}
+
+func doInit(conf *config.CliConfig, force bool, includeAll bool) error {
+	var err error
+	if len(conf.Modules) == 0 || force {
+		conf.Modules, err = findModules(".")
+		if err != nil {
+			initLogger.Warningf("A non-fatal error occured during module discovery: %s", err.Error())
+		}
+
+		if !includeAll {
+			// fitler suspicious modules
+			filteredModuleConfigs := []config.ModuleConfig{}
+			for _, c := range conf.Modules {
+				if matched, err := regexp.MatchString("(docs?/|test|example|vendor/)", c.Path); err != nil || matched != true {
+					filteredModuleConfigs = append(filteredModuleConfigs, c)
+				} else {
+					initLogger.Warningf("Filtering out suspcious module: %s (%s)", c.Name, c.Path)
+				}
+			}
+			conf.Modules = filteredModuleConfigs
+		}
+	} else {
+		initLogger.Warningf("%d module(s) found in config file (`%s`); skipping initialization.", len(conf.Modules), conf.ConfigFilePath)
+	}
+	return nil
+}
+
+// findModules calls DiscoverModules() on all available integrations and returns a new config state
+// to implement fail-open behavior, do not handle the err returned by this function
```

Comment:
That's not what fail-open means, get rid of this part of the comment.

Reply:
What do you mean?

Reply:
I mean, in reference to "to implement fail-open behavior, do not handle the err returned by this function":

1. "Fail-open" is only meaningful in the context of security, and doesn't mean anything here. It would be more accurate to say "fail-safe" but that's not quite accurate either.
2. Everybody knows they can ignore errors, this part of the comment adds no new information and is potentially confusing.